File: DKS-API/DTOs/SDevDtrVsList.cs
using System;
using DKS_API.Helpers;

namespace DKS_API.DTOs
{
    public class SDevDtrVsList : PaginationParams
    {
        public string Season { get; set; }
        public string Article { get; set; }
        public string ModelNo { get; set; }
        public string ModelName { get; set; }
        public string DeveloperId { get; set; }
        public string DevTeamId { get; set; }
    }
}

File: DKS-SPA/src/app/core/_models/s-dev-dtr-vs-list.ts
import { Pagination } from "./pagination";

export class SDevDtrVsList extends Pagination {

  article: string;
  season: string;
  modelNo: string;
  modelName: string;
  developerId: string;
  devTeamId: string;

  constructor() {
    super();
    this.isPaging = true; // 不開分頁
    this.article ="";
    this.season ="";
    this.modelNo ="";
    this.modelName ="";
    this.developerId ="";
    this.devTeamId ="";
  }
  
  public setPagination(pagination: Pagination) {
    this.currentPage = pagination.currentPage;
    this.itemsPerPage = pagination.itemsPerPage;
    this.totalItems = pagination.totalItems;
    this.totalPages = pagination.totalPages;
  }
}
